### DISCUSSION TITLE
Berry: Using web_send

### DISCUSSION THREAD

**Discussion by msetina** [arendst/Tasmota#19593] [Q&A]
I am trying to find documentation for web_send regarding special tags in msg.
I found {s},{m} and {e} in an example. 
"{s}MPU6886 acc_x{m}%.3f G{e}"

The example goes like this:
**#- display sensor value in the web UI -#
    def web_sensor()
      import string
      var msg = string.format(
               "{s}MPU6886 acc_x{m}%.3f G{e}"..
               "{s}MPU6886 acc_y{m}%.3f G{e}"..
               "{s}MPU6886 acc_z{m}%.3f G{e}"..
               "{s}MPU6886 gyr_x{m}%i dps{e}"..
               "{s}MPU6886 gyr_y{m}%i dps{e}"..
               "{s}MPU6886 gyr_z{m}%i dps{e}",
                self.accel[0], self.accel[1], self.accel[2], self.gyro[0], self.gyro[1], self.gyro[2])
      tasmota.web_send_decimal(msg)
    end**

**Comment by sfromis** [comment]
You can read `{s}` as start, `{m}` as middle and `{e}` as end. They are used for aligning the sensor names and values into two columns on the Tasmota front page. There are no other similar tags.

They are described in this section: https://tasmota.github.io/docs/Berry-Cookbook/#creating-an-i2c-driver right above the text you copied here.

**Reply by barbudor** [reply]
There is also `{t]` for table
They are actually shortcuts that are replaced by HTML at run time by Javascipt in the browser (in order to save ESP flash)
The actual replacement is defined here:
https://github.com/search?q=repo%3Aarendst%2FTasmota+path%3A%2F%5Etasmota%5C%2Fhtml_uncompressed%5C%2F%2F+%2F%5C%7B%5Cw%5C%7D%2F&type=code

**Reply by msetina** [reply]
Thanks. Usefull, but @sfromis, I did not find before. Went direct to full example.
